File: session-06-capstone/python/resilience/fallback.py
from kitchen.service import KitchenService, Preparation


class FallbackKitchenService(KitchenService):
    def __init__(self, inner: KitchenService):
        self.inner = inner

    def prepare(self, order_id: int, drink: str) -> Preparation:
        try:
            return self.inner.prepare(order_id, drink)
        except Exception as e:
            print(f"[Fallback] Kitchen failed: {e} -- returning queued for order #{order_id}")
            return Preparation(order_id=order_id, drink=drink, status="queued")

File: session-06-capstone/go/resilience/fallback.go
package resilience

import (
	"fmt"

	"coffeeshop-capstone/kitchen"
)

type FallbackKitchenService struct {
	Inner kitchen.KitchenService
}

func (s *FallbackKitchenService) Prepare(orderID int, drink string) (*kitchen.Preparation, error) {
	prep, err := s.Inner.Prepare(orderID, drink)
	if err != nil {
		fmt.Printf("[Fallback] Kitchen failed: %s -- returning queued for order #%d\n", err, orderID)
		return &kitchen.Preparation{OrderID: orderID, Drink: drink, Status: "queued"}, nil
	}
	return prep, nil
}
